src/extension/inpage-script/index.js

```diff
@@ -4,87 +4,109 @@ import LiquidProvider from "~/extension/providers/liquid";
 import NostrProvider from "~/extension/providers/nostr";
 import WebBTCProvider from "~/extension/providers/webbtc";
 import WebLNProvider from "~/extension/providers/webln";
+import shouldInjectInpage from "./shouldInject";
 
-if (document) {
-  // Intercept any `lightning:` requests
+function init() {
+  if (document) {
+    const inject = shouldInjectInpage();
+    if (!inject) return;
+    window.liquid = new LiquidProvider();
+    window.alby = new AlbyProvider();
+    window.nostr = new NostrProvider();
+    window.webbtc = new WebBTCProvider();
+    window.webln = new WebLNProvider();
 
-  window.liquid = new LiquidProvider();
-  window.alby = new AlbyProvider();
-  window.nostr = new NostrProvider();
-  window.webbtc = new WebBTCProvider();
-  window.webln = new WebLNProvider();
+    const readyEvent = new Event("webln:ready");
+    window.dispatchEvent(readyEvent);
+    // Intercept any `lightning:` requests
+    window.addEventListener(
```

Comment:
Can we move this to a function like "registerLightningLinkClickHandler"?

Reply:
or may be just, lightningClickHandler?